### ISSUE TITLE
SimplifyTypeNames requires two simplification passes

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#40647]
Given: 

```c#
public void Goo(object o1, object o2) => Object.ReferenceEquals(o1, o2);
```

SimplifyTypeNames will first simplify this to:

```c#
public void Goo(object o1, object o2) => object.ReferenceEquals(o1, o2);
```

Then another pass will produce

```c#
public void Goo(object o1, object o2) => ReferenceEquals(o1, o2);
```

It should just produce the latter in one pass.

**Comment by sharwell** [comment]
We should report both IDE0001 here (fix produces the final result directly) and IDE0049 here (fix produces the second result). Users may have different severities for these two diagnostics and/or have one of them disabled.

**Comment by CyrusNajmabadi** [comment]
@sharwell I think I agree.  As i continue working in tihs area, I'll investigate and beef up testing.

**Comment by CyrusNajmabadi** [comment]
note: looking into this, it would be somewhat complex to support. the current pipeline is very oriented around producing a single result (or nothing).
